ntp-proto/src/source.rs

```diff
@@ -637,14 +653,16 @@ impl<Controller: SourceController> NtpSource<Controller> {
                 let tries_left = tries_left.saturating_sub(1);
                 if message.is_upgrade() {
                     debug!("Received a valid upgrade response, switching to NTPv5!");
-                    self.protocol_version = ProtocolVersion::V5;
+                    self.protocol_version = ProtocolVersion::UpgradedToV5;
                 } else if tries_left == 0 {
                     debug!("Server does not support NTPv5, stopping the upgrade process");
                     self.protocol_version = ProtocolVersion::V4;
                 } else {
                     debug!(tries_left, "Server did not yet respond with upgrade code");
                     self.protocol_version = ProtocolVersion::V4UpgradingToV5 { tries_left };
                 };
+            } else if let ProtocolVersion::UpgradedToV5 = self.protocol_version {
+                self.protocol_version = ProtocolVersion::V5;
```

Comment:
Do we need to check if the version in the packet we just got is actually v5 before we commit here?